client/src/app/users/user-list.component.html

```diff
@@ -4,36 +4,49 @@
       <mat-card-title id="user-list-title">Users</mat-card-title>
       <mat-divider></mat-divider>
       <mat-card-content>
+        <label for="userName">Name:</label>
         <mat-form-field>
           <input
             matInput
             id="userName"
             placeholder="Filter by name"
             (input)="updateName($event.target.value)">
         </mat-form-field>
+        <br>
+        <!--this has an annoying error about setSelctionRange that we should probably deal with-->
+        <label for="userAge">Age (filter in Angular):</label>
         <mat-form-field>
           <input
             matInput
             id="userAge"
             type="number"
             placeholder="Filter by age"
+            min="0" max="200"
             (input)="updateAge($event.target.value)">
         </mat-form-field>
         <!--<mat-form-field>-->
         <!--<input matInput id="userCompany" #input type="text" placeholder="Search for company" (input)="contentChange($event.target.value)">-->
         <!--</mat-form-field>-->
 
-        <mat-accordion *ngIf="users; else usersError">
-          <mat-expansion-panel #elem [id]="user.email" *ngFor="let user of this.filteredUsers">
-            <mat-expansion-panel-header>
+        <!--example of filtering on the server-->
+        <mat-form-field>
+          <mat-label>Age (filter on server):</mat-label>
+          <mat-select (selectionChange)="updateServerFilterAge($event.value)" [(ngModel)]="userAgeForServerFilter" id="userAgeForServerFilter">
```

Comment:
Something like age should be a number input rather than a limited selection like this.

Reply:
Good point. We have limited data to choose from, and I chose some interesting numbers. If we had more data, I can imagine it would be possible to come up with a better use of a dropdown. Perhaps company name would be a better filter to use in this manner? I wanted there to be something that we could use a dropdown for to show another type of input, and I wanted there to be some options that had interesting results.

The point of this change was to show filtering on the server and to give some way for that to happen.

Reply:
Do we want to replace the age filters with a selection of company names for this lab? There might be a few places to change because the age selection served as the only example of filtering on the server.

Reply:
Let's leave it for now. Perhaps we should make an issue for the new idea.